packages/turbo-workspaces/src/getWorkspaceDetails.ts

```diff
@@ -0,0 +1,29 @@
+import { ConvertError } from "./errors";
+import managers from "./managers";
+import { Project } from "./types";
+import { directoryInfo } from "./utils";
+
+export default async function getWorkspaceDetails({
+  root,
+}: {
+  root: string;
+}): Promise<Project> {
+  const { exists, absolute: workspaceRoot } = directoryInfo({
+    directory: root,
+  });
+  if (!exists) {
+    throw new ConvertError(
+      `Could not find directory at ${workspaceRoot}. Ensure the directory exists.`
+    );
+  }
+
+  for (const { detect, read } of Object.values(managers)) {
+    if (await detect({ workspaceRoot })) {
```

Comment:
I'm not sure that this is identical with what we do elsewhere. (`turbo`, `vc`)

This checks for lockfile, and then `package.json` on a per-package basis. I think we usually check `package.json` as a first pass, and then try to deduce from lockfile presence second in yarn > npm > pnpm order.

Can you confirm? This note is from memory.

Reply:
Yep you're right, we used to do the opposite in turbo but swapped once we started to enforce the package manager value in 1.1, I'll invert this